### ISSUE TITLE
ngx-stripe-card is not a known element

### ISSUE THREAD

**Issue by knowledgeplaces** [richnologies/ngx-stripe#163]
Same problem has already been reported on https://github.com/richnologies/ngx-stripe/issues/115 but I have still the issue even adding the NgxStripeModule.forChild() in app.module.ts.

Here is the component template code except:
```
<div fxLayout="row" fxLayoutAlign="start">
<ngx-stripe-card [options]="cardOptions" [elementsOptions]="elementsOptions"></ngx-stripe-card>
<button i18n mat-raised-button color="primary" type="button"
	(click)="this.testStripeApi()">Test Stripe API</button>
</div>
```
Where I have the error:
'ngx-stripe-card' is not a known element:

Here is my component code:
```
// stripe imports
import {
	StripeCardComponent,
	StripeInstance,
	StripeFactoryService
} from 'ngx-stripe';
import {
	StripeCardElementOptions,
	StripeElementsOptions
} from '@stripe/stripe-js';

@Component({
	selector: 'app-eshop-settings-form',
	templateUrl: './eshop-settings-form.component.html',
	styleUrls: ['./eshop-settings-form.component.css']
})
export class EshopSettingsFormComponent implements OnInit, AfterViewInit {

	@ViewChild(StripeCardComponent) card: StripeCardComponent;
	cardOptions: StripeCardElementOptions = {
		style: {
			base: {
				iconColor: '#666EE8',
				color: '#31325F',
				fontWeight: '300',
				fontFamily: '"Helvetica Neue", Helvetica, sans-serif',
				fontSize: '18px',
				'::placeholder': {
					color: '#CFD7E0'
				}
			}
		}
	};
	elementsOptions: StripeElementsOptions = {
		locale: 'en'
	};
```
Code of my test button:
```
	testStripeApi(): void {
		const name = this.eShopSettingsFormGroup.get('stripeCardTest').value;
		this.stripe.createToken(this.card.element, { name }).subscribe((result) => {
			if (result.token) {
				// Use the token
				console.log(result.token.id);
			} else if (result.error) {
				// Error creating the token
				console.log(result.error.message);
			}
		});
	}
```
Any advice appreciated.

**Comment by richnologies** [comment]
Hi @knowledgeplaces, thanks for creating a separate issue, is easier for me to keep track this way.

`forChild()` is not necessary. I simple add it once because this kind of issues, but right now is not recommended (actually I will probably add the deprecated warning after this)

I prepare for you a Stackblitz example with using your snippets: https://stackblitz.com/edit/ngx-stripe-issue-163

A few notes:

- The recommended way is to use `forRoot` in the main module, either app.module or core.module if you have something like that for your providers. 
- As with any other library, you need to add the NgxStripeModule as in import in any module where you are going to use it, in the example that would be the FeatureModule, but here you don't need neither `forRoot()` not `forChild()`

I would like to apologize for the confusion.

If this helps you please let me know so I can adapt the docs to help other. If not, you can create a free account in Stackblitz and make any changes you need to replicate the error, that would help me fix it quickly. 

If you're not able to replicate or you have anymore questions, do not hesitate to ask, I'm here to help :)

Kind regards,

R

**Comment by knowledgeplaces** [comment]
Thanks for quick feedback, but I am not lucky yet.
I have followed your sample code.
In app.module.ts, I have no error with:
`
import { NgxStripeModule } from 'ngx-stripe';
`
and
`
NgxStripeModule.forRoot('pk_test_***my_key***')
`
and
`
			{
				path: 'stripetest',
				loadChildren: () =>
					import('./stripe-test-form/stripe-test-form.module').then((m) => m.StripeTestFormModule),
			},`
in stripe-test-form.module.ts, no error:
`
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';

import { NgxStripeModule } from 'ngx-stripe';

import { StripeTestFormComponent } from './stripe-test-form.component';

@NgModule({
	declarations: [StripeTestFormComponent],
	imports: [
		CommonModule,
		RouterModule.forChild([
			{
				path: '',
				component: StripeTestFormComponent,
			},
		]),
		ReactiveFormsModule,
		NgxStripeModule,
	],
})
export class FeatureModule { }

`

**Comment by richnologies** [comment]
This is odd. What versions are you using of:

- Angular
- Typescript
- NodeJS
- ngx-stripe

**Comment by richnologies** [comment]
Hi @knowledgeplaces, glad to hear your are making progress. 

It seems to be something is odd with webpack. Maybe I would suggest to use ngx-stripe version 12 to match the Angular one, but to be honest, I don't is going to have any impact, is "just in case". I will try it with your versions to see if I can replicate the error. 

In the meantime, for the dynamic key, we have the idea of a factory service, check the docs here: 
https://docs.ngx-stripe.dev/core-concepts/service-factory

I've updated the stackblitz example to use the factory service: https://stackblitz.com/edit/ngx-stripe-issue-163

In summary:

- We don't pass the key to the `forRoot` method because we don't know it yet. 
- We don't use the Stripe Factory directly, but we create a StripeInstance using the StripeFactory
- This instance has the same API as the StripeService. We need to pass the instance to the <ngx-stripe-card> element too.

**Comment by knowledgeplaces** [comment]
Hi @richnologies ,

With your updated Stackblitz, I have been able to add support for dynamic key and it works perfectly, thanks.

I have still the error  'ngx-stripe-card' is not a known element:' when I edit the component template file, but the app compiles with no errors and runs file, and when I close my Eclipse IDE and reopen it, the file is not marked in error, so I will keep everything as is.

Thanks for your quick and efficient support on this issue.

**Comment by angiexroap** [comment]
@richnologies ... hi sir, recently stripe js updated to 1.50 version, ngx stripe has problem with one of its classes, StripeElementsOptions, now this name is BasicStripeElementsOptions, all modules broke with that

 "@angular/core": "^15.1.0",
 "@angular/cli": "~15.1.2",
    "@angular/compiler-cli": "^15.1.0",
 "ngx-stripe": "^15.5.1",
 "@stripe/stripe-js": "^1.50.0"
 
 check it please sir, thanks

**Comment by richnologies** [comment]
Hey @angiexroap, can you please update `@stripe/stripe-js` to version `1.51.0` and let know if that works for you?

**Comment by angiexroap** [comment]
hi sir, ya, all its ok
